frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx

```diff
@@ -0,0 +1,117 @@
+import { IconUpload, IconX } from '@posthog/icons'
+import { LemonButton, LemonDivider, LemonFileInput, LemonModal, LemonSkeleton, lemonToast } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { useUploadFiles } from 'lib/hooks/useUploadFiles'
+import { useState } from 'react'
+
+import { experimentLogic } from '../experimentLogic'
+import { VariantTag } from './components'
+
+export function VariantPreviewImage({
+    variantKey,
+    rolloutPercentage,
+}: {
+    variantKey: string
+    rolloutPercentage: number
+}): JSX.Element {
+    const { experiment } = useValues(experimentLogic)
+    const { updateExperimentVariantImages } = useActions(experimentLogic)
+
+    const [mediaId, setMediaId] = useState(experiment.parameters?.variant_preview_media_ids?.[variantKey] || null)
+    const [isLoadingImage, setIsLoadingImage] = useState(true)
+    const [isModalOpen, setIsModalOpen] = useState(false)
+
+    const { setFilesToUpload, filesToUpload, uploading } = useUploadFiles({
+        onUpload: (_, __, id) => {
+            setMediaId(id)
+            if (id) {
+                const updatedVariantImages = {
+                    ...experiment.parameters?.variant_preview_media_ids,
+                    [variantKey]: id,
+                }
+                updateExperimentVariantImages(updatedVariantImages)
+            }
+        },
+        onError: (detail) => {
+            lemonToast.error(`Error uploading image: ${detail}`)
+        },
+    })
+
+    return (
+        <div className="space-y-4">
+            {!mediaId ? (
+                <LemonFileInput
+                    accept="image/*"
+                    multiple={false}
+                    onChange={setFilesToUpload}
+                    loading={uploading}
+                    value={filesToUpload}
+                    callToAction={
+                        <div className="relative">
+                            <div className="Lettermark Lettermark--xlarge Lettermark--rounded Lettermark--variant-4">
+                                <IconUpload />
+                            </div>
+                        </div>
```

Comment:
Plugging into the internals of another component is fragile – changes to Lettermark styling in the future are likely to break `VariantPreviewImage`. On a different note, there is very poor contrast between the upload icon and its background.

I propose a more conventional file input, which also lets us be more descriptive about this preview feature:
```suggestion
                            <>
                                <IconUpload className="text-2xl" />
                                <span>Upload a preview of this variant's UI</span>
                            </>
```

Looking like this:
<img width="279" alt="Screenshot 2024-10-07 at 12 05 34" src="https://github.com/user-attachments/assets/f4014a4d-c913-4ac4-b6e6-c933490bf481">